test/FunctionalTests/Framework/Util/FrameworkUtil.cs

```diff
@@ -164,8 +164,17 @@ private static string FindEnlistmentRootHeuristically()
                 return result;
             }
 
-            // Second use current assembly location to determine.
+            // Second, use environment variable defined in build agents
+            // See: https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables
+            result = Environment.GetEnvironmentVariable("BUILD_SOURCESDIRECTORY");
+            if (!string.IsNullOrEmpty(result))
+            {
+                return result;
+            }

```

Comment:
It's unlikely that we're going to revert to the AzDO pipelines any time soon. I think you should move this block to the top of the method